src/client/app/redux/selectors/adminSelectors.ts

```diff
@@ -0,0 +1,309 @@
+		defaultHelpUrl: adminState.defaultHelpUrl
+	})
+)
+
+
+export const selectPossibleGraphicUnits = createAppSelector(
+	selectUnitDataById,
+	unitDataById => potentialGraphicUnits(unitDataById)
+)
+
+/**
+ * Calculates the set of all possible meter units for a meter.
+ * This is any unit that is of type unit or suffix.
+ * @returns The set of all possible graphic units for a meter
+ */
+export const selectPossibleMeterUnits = createAppSelector(
+	selectAllUnits,
+	unitData => {
+		let possibleMeterUnits = new Set<UnitData>();
+		// The meter unit can be any unit of type meter.
+		unitData.forEach(unit => {
+			if (unit.typeOfUnit == UnitType.meter) {
+				possibleMeterUnits.add(unit);
+			}
+		});
+		// Put in alphabetical order.
+		possibleMeterUnits = new Set(_.sortBy(Array.from(possibleMeterUnits), unit => unit.identifier.toLowerCase(), 'asc'));
+		// The default graphic unit can also be no unit/-99 but that is not desired so put last in list.
+		return possibleMeterUnits.add(noUnitTranslated());
+	}
+)
+
+export const selectUnitName = createAppSelector(
+	// This is the unit associated with the meter.
+	// The first test of length is because the state may not yet be set when loading. This should not be seen
+	// since the state should be set and the page redrawn so just use 'no unit'.
+	// The second test of -99 is for meters without units.
+	// ThisSelector takes an argument, due to one or more of the selectors accepts an argument (selectUnitWithID selectMeterDataWithID)
+	selectUnitDataById,
+	selectMeterById,
+	(unitDataById, meterData) => {
+		const unitName = (Object.keys(unitDataById).length === 0 || !meterData || meterData.unitId === -99) ?
+			noUnitTranslated().identifier : unitDataById[meterData.defaultGraphicUnit]?.identifier
+		return unitName
+	}
+)
+
+export const selectGraphicName = createAppSelector(
+	// This is the default graphic unit associated with the meter. See above for how code works.
+	// notice that this selector is written with inline selectors for demonstration purposes
+	selectUnitDataById,
+	selectMeterById,
+	(unitDataById, meterData) => {
+		const graphicName = (Object.keys(unitDataById).length === 0 || !meterData || meterData.defaultGraphicUnit === -99) ?
+			noUnitTranslated().identifier : unitDataById[meterData.defaultGraphicUnit].identifier
+		return graphicName
+	}
+)
+
+
+export const selectGraphicUnitCompatibility = createAppSelector(
+	[
+		selectPossibleGraphicUnits,
+		selectPossibleMeterUnits,
+		(_state, meterDetails: MeterData) => meterDetails.unitId,
+		(_state, meterDetails: MeterData) => meterDetails.defaultGraphicUnit
+	],
+	(possibleGraphicUnits, possibleMeterUnits, unitId, defaultGraphicUnit) => {
+		// Graphic units compatible with currently selected unit
+		const compatibleGraphicUnits = new Set<UnitData>();
+		// Graphic units incompatible with currently selected unit
+		const incompatibleGraphicUnits = new Set<UnitData>();
+		// If unit is not 'no unit'
+		if (unitId != -99) {
```

Comment:
This is used for both edit and create. For create there used to be a separate test for -999. I'm noting since this may need to change to get the other code to allow no unit.

Reply:
I think this is resolved. Could @ChrisMart21 verify.

Reply:
Is resolved with recent create meter modal updates.